specs/_features/eip7732/beacon-chain.md

```diff
@@ -0,0 +1,716 @@
+- A full block for the current slot (both the proposer and the builder revealed on time). 
+
+## Constants
+
+### Payload status
+
+| Name | Value | 
+| - | - | 
+| `PAYLOAD_ABSENT` | `uint8(0)` |
+| `PAYLOAD_PRESENT` | `uint8(1)` | 
+| `PAYLOAD_WITHHELD` | `uint8(2)` | 
+| `PAYLOAD_INVALID_STATUS` | `uint8(3)` |
+
+## Preset
+
+### Misc
+
+| Name | Value | 
+| - | - | 
+| `PTC_SIZE` | `uint64(2**9)` (=512)  # (New in EIP-7732) |
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_BEACON_BUILDER`     | `DomainType('0x1B000000')`  # (New in EIP-7732)|
+| `DOMAIN_PTC_ATTESTER`       | `DomainType('0x0C000000')`  # (New in EIP-7732)|
+
+### Max operations per block
+
+| Name | Value |
+| - | - |
+| `MAX_PAYLOAD_ATTESTATIONS` | `2**2` (= 4)  # (New in EIP-7732) |
+
+## Containers
+
+### New containers
+
+#### `PayloadAttestationData`
+
+```python
+class PayloadAttestationData(Container):
+    beacon_block_root: Root
+    slot: Slot
+    payload_status: uint8
+```
+
+#### `PayloadAttestation`
+
+```python
+class PayloadAttestation(Container):
+    aggregation_bits: Bitvector[PTC_SIZE]
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+#### `PayloadAttestationMessage`
+
+```python
+class PayloadAttestationMessage(Container):
+    validator_index: ValidatorIndex
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+#### `IndexedPayloadAttestation`
+
+```python
+class IndexedPayloadAttestation(Container):
+    attesting_indices: List[ValidatorIndex, PTC_SIZE]
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+#### `SignedExecutionPayloadHeader`
+
+```python
+class SignedExecutionPayloadHeader(Container):
+    message: ExecutionPayloadHeader
+    signature: BLSSignature
+```
+    
+#### `ExecutionPayloadEnvelope`
+
+```python
+class ExecutionPayloadEnvelope(Container):
+    payload: ExecutionPayload
+    builder_index: ValidatorIndex
+    beacon_block_root: Root
+    blob_kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    payload_withheld: boolean
+    state_root: Root
+```
+
+#### `SignedExecutionPayloadEnvelope`
+
+```python
+class SignedExecutionPayloadEnvelope(Container):
+    message: ExecutionPayloadEnvelope
+    signature: BLSSignature
+```
+
+### Modified containers
+
+#### `BeaconBlockBody`
+
+**Note:** The Beacon Block body is modified to contain a `Signed ExecutionPayloadHeader`. The containers `BeaconBlock` and `SignedBeaconBlock` are modified indirectly.
+
+```python
+class BeaconBlockBody(Container):
+    randao_reveal: BLSSignature
+    eth1_data: Eth1Data  # Eth1 data vote
+    graffiti: Bytes32  # Arbitrary data
+    # Operations
+    proposer_slashings: List[ProposerSlashing, MAX_PROPOSER_SLASHINGS]
+    attester_slashings: List[AttesterSlashing, MAX_ATTESTER_SLASHINGS]
+    attestations: List[Attestation, MAX_ATTESTATIONS]
```

Comment:
Why not `MAX_ATTESTATIONS_ELECTRA`? Do we need 128 once more?

Reply:
No, I think this is a bug